datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -580,7 +583,7 @@ fn coerced_from<'a>(
         {
             Some(type_into.clone())
         }
-        (Interval(_), _) if matches!(type_from, Utf8 | LargeUtf8) => {
+        (Interval(_), _) if matches!(type_from, Utf8 | LargeUtf8 | Null) => {
```

Comment:
This is what I want to deprecate but not yet, at least we don't make it more complex

Reply:
rollback codes in `functions.rs`